src/frontend/Komponenter/Behandling/Brev/FrittståendeBrev.tsx

```diff
@@ -53,38 +48,61 @@ const ManueltBrev: React.FC<Props> = ({ fagsakId }) => {
 
     const [overskrift, settOverskrift] = useState('');
     const [avsnitt, settAvsnitt] = useState<IAvsnitt[]>(førsteRad);
-    const [brev, settBrev] = useState<Ressurs<string>>(byggTomRessurs());
-    const { axiosRequest, innloggetSaksbehandler } = useApp();
+    const { axiosRequest } = useApp();
 
-    const personopplysningerConfig: AxiosRequestConfig = useMemo(
+    const personopplysningerFagsakConfig: AxiosRequestConfig = useMemo(
         () => ({
             method: 'GET',
-            url: `/familie-ef-sak/api/personopplysninger/fagsak/${fagsakId}`,
+            url: `/familie-ef-sak/api/personopplysninger/fagsak/${props.fagsakId}`,
         }),
-        [fagsakId]
+        [props.fagsakId]
     );
 
+    const personopplysningerBehandlingConfig: AxiosRequestConfig = useMemo(
+        () => ({
+            method: 'GET',
+            url: `/familie-ef-sak/api/personopplysninger/behandling/${props.behandlingId}`,
+        }),
+        [props.behandlingId]
+    );
+
+    const personopplysningerConfig = props.fagsakId
+        ? personopplysningerFagsakConfig
+        : personopplysningerBehandlingConfig;
+
     const personopplysninger = useDataHenter<IPersonopplysninger, null>(personopplysningerConfig);
 
     const genererBrev = () => {
         if (personopplysninger.status !== RessursStatus.SUKSESS) return;
-
-        const brevdato = dagensDatoFormatert();
-
-        axiosRequest<string, IManueltBrev>({
-            method: 'POST',
-            url: `/familie-ef-sak/api/manueltbrev`,
-            data: {
-                overskrift,
-                avsnitt,
-                saksbehandlersignatur: innloggetSaksbehandler?.displayName || 'Ukjent',
-                brevdato,
-                ident: personopplysninger.data.personIdent,
-                navn: personopplysninger.data.navn.visningsnavn,
-            },
-        }).then((respons: Ressurs<string>) => {
-            settBrev(respons);
-        });
+        const { fagsakId, behandlingId } = props;
+
+        if (fagsakId) {
+            axiosRequest<string, IFrittståendeBrev>({
+                method: 'POST',
+                url: `/familie-ef-sak/api/brev/frittstående`,
+                data: {
+                    overskrift,
+                    avsnitt,
+                    fagsakId,
+                },
+            }).then((respons: Ressurs<string>) => {
+                if (props.oppdaterBrevressurs) props.oppdaterBrevressurs(respons);
+            });
+        }
+
+        if (behandlingId) {
+            axiosRequest<string, IFrittståendeBrev>({
+                method: 'POST',
+                url: `/familie-ef-sak/api/brev/fritekst`,
+                data: {
+                    overskrift,
+                    avsnitt,
+                    behandlingId,
+                },
+            }).then((respons: Ressurs<string>) => {
+                if (props.oppdaterBrevressurs) props.oppdaterBrevressurs(respons);
+            });
+        }
```

Comment:
Risikerer vi at brevet genereres to ganger dersom `fagsakId` og `behandlingId` begge er sendt med som props? Kanksje vurdere en else, eller kan det tenkes at vi ønsker å kalle begge endepunktene?

Reply:
Fikset i 2690ebf